src/components/Map/utils/utils.js

```diff
@@ -0,0 +1,223 @@
+	poi: 'poi',
+	transit: 'transit',
+	administrative: 'administrative',
+	landscape: 'landscape',
+	road: 'road',
+	water: 'water',
+	_default: 'poi'
+};
+
+export const getMapStylers = (rules) => {
+	if (!rules) return [];
+
+	const styleRules = [];
+	const allRules = Array.isArray(rules) ? rules : [rules];
+
+	allRules.forEach((rule) => {
+		const featureType = mapFeatureTypes[rule] || mapFeatureTypes._default; // eslint-disable-line no-underscore-dangle
+		const { styler, value } = mapStylers._default; // eslint-disable-line no-underscore-dangle
+
+		const ruleObj = {
+			featureType,
+			stylers: [{ [styler]: value }]
+		};
+
+		styleRules.push(ruleObj);
+	});
+
+	return styleRules;
+};
+
+export const formatCoordinates = (latitude, longitude) => {
+	const direction = {
+		vehicleStopover: false,
+		sideOfRoad: false,
+		location: {
+			latLng: {
+				latitude,
+				longitude
+			}
+		}
+	};
+	return direction;
+};
+
+/**
+ * takes an array of waypoints and returns them parsed
+ *
+ * @param {array} waypoints
+ * @return {array}
+ */
+export const createLatLngObjectsFromArray = (waypoints) => {
+	if (!waypoints || !Array.isArray(waypoints) || !waypoints.length) return [];
+
+	return waypoints
+		.map((waypoint) =>
+			waypoint.position.lng && waypoint.position.lat
+				? formatCoordinates(waypoint.position.lat, waypoint.position.lng)
+				: null
+		)
+		.filter(Boolean);
+};
+
+/**
+ *
+ * @param {object} lock
+ * @returns {boolean}
+ */
+export const validLock = (lock) => !!lock && isObject(lock) && !!Object.keys(lock).length;
+
+const parseProp = (prop) => (typeof prop === 'object' ? prop : { icon: prop });
+
+export const addOriginAndDestination = (originMarker, destinationMarker, index, markersArray) => {
+	const destinationValidation =
+		index === markersArray.length - 1 ? { ...parseProp(destinationMarker) } : {};
+	const markerProps = index === 0 ? { ...parseProp(originMarker) } : destinationValidation;
+
+	return markerProps;
+};
+
+export const findSelectedPosition = (selectedRows, marker) => {
+	const index =
+		marker.type === 'warehouse'
+			? selectedRows.indexOf(marker.id)
+			: selectedRows.indexOf(marker.shippingId);
+	return index !== -1 ? index + 1 : null;
+};
+
+/* Maps every marker to extend the map's bounds so they can all be initially shown */
+export const getBounds = (markers) => {
+	const bounds = new window.google.maps.LatLngBounds();
+	markers.map((marker) => {
+		const { lat, lng } = marker;
+		return bounds.extend(new window.google.maps.LatLng(lat, lng));
+	});
+
+	return bounds;
+};
+
+/* If there are several markers, updates the map's bounds to show them all, otherwise, the map keeps its original position */
+export const showAllMarkers = (map, markers, centerCoordinate) => {
+	if (markers.length) {
+		const markersFlatted = markers.flatMap((marker) =>
+			marker.points.map((point) => point.position)
+		);
+		map.fitBounds(getBounds(markersFlatted));
+		map.setCenter(centerCoordinate);
```

Comment:
@FranciscoZapa aca esta función esta tirando un error
`main.js:164 InvalidValueError: setCenter: not a LatLng or LatLngLiteral with finite coordinates: in property lat: NaN is not an accepted value
`
me parece que deberías validar `lat` y `lng` antes de setear el mapa ya que están llegando en `Nan`